app/Kernel.php

```diff
@@ -5,6 +5,8 @@
  * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
  */
 
+declare(strict_types=1);
```

Comment:
shouldnt they come directly after <?php ?

Reply:
It's the question that I wanted to discuss with you :) Most files where it is used, have it after license block, that's why I added also after license. In some other it stays before license. So we need some standard for that. For me it looks also more natural when it is used right after <?php 

So do we agree that it should stay right after <?php ?

Reply:
Lets add a sniff then that ensures and auto fixes this please.
You can test the sniff on this PR ;)

Reply:
I had an another PR for Code Sniffer - https://github.com/spryker-sdk/upgrader/pull/72 

But no problem, I extend it I add it here

Reply:
@dereuromark , according to PCR-12 strict types declaration should go after file-level documentation: https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements

So according to PCR12 it should really look like: 

`<?php

/**
 * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
 * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
 */

declare(strict_types=1);`

So I implement it according to PCR12 specification, ok?

Reply:
I see

Reply:
@dereuromark , so it's not so easy to implement sniffer that will work exactly according to PSR12 :)

We can use SlevomatCodingStandard.TypeHints.DeclareStrictTypesSniff (https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/TypeHints/DeclareStrictTypesSniff.php) 

But it was developed to add strict_types right after <?php and according to my tests it has some bugs. So we can use it, it will add lines before and after declare(strict_types=1); if it already exists after file doc. If it doesn't exist at all, this sniffer will add it right after <?php (before file doc), so we get again 2 standards.

I made a dirty rewrite of that sniffer to have a look what I can do with that:
https://github.com/spryker/code-sniffer/compare/master...feature/PPLUS-1585

So the new one can add declare(strict_types=1); on the right place (after file doc) if it doesn't exist in file, can add the lines before and after it. But it still has problem that it can't remove declare(strict_types=1); if it already exists in the file after <?php

So if I spend more time on our own sniffer implementation, I should also add that remove functionality. So just a question - do we want to have our own sniffer implementation for that? Or should we just use some existing one with known issues? Or do we want to use declare(strict_types=1); after <?php ignoring PSR12?

Reply:
We can have our own if it works correctly then and doesnt need too much time to build

Reply:
Ok, I did it: https://github.com/spryker/code-sniffer/pull/356/files

So I implemented one sniffer that removes the declare(strict_types=1); if it is directly after <?php

and implemented one sniffer that add it after file doc, if it is not there. I tested it on our projects and it works as expected.